ckanext/unaids/tests/test_plugin.py

```diff
@@ -42,52 +42,85 @@ class TestResourceLastModified(object):
     whenever a resource is created or edited.
     '''
 
-    URL_TYPE_FILE_UPLOAD = 'upload'
-    URL_TYPE_LINK = ''
-
-    def test_resource(self, url_type):
-        resource_dict = {'url_type': url_type}
-        if url_type == self.URL_TYPE_FILE_UPLOAD:
-            resource_dict.update({
-                'url': 'file.csv',
-                'lfs_prefix': 'prefix123',
-                'sha256': 'sha256',
-                'size': 100,
-            })
-        elif url_type == self.URL_TYPE_LINK:
-            resource_dict.update({
-                'url': 'http://example.com'
-            })
-        return resource_dict
-
-    def test_null_to_file(self):
-        resource = self.test_resource(self.URL_TYPE_FILE_UPLOAD)
-        _update_resource_last_modified_date(resource=resource)
-        assert 'last_modified' in resource
-
-    def test_file_to_null(self):
-        resource = self.test_resource(self.URL_TYPE_FILE_UPLOAD)
-        _update_resource_last_modified_date({}, resource)
-        assert 'last_modified' not in resource
-
-    def test_file_to_url(self):
-        current = self.test_resource(self.URL_TYPE_FILE_UPLOAD)
-        resource = self.test_resource(self.URL_TYPE_LINK)
-        _update_resource_last_modified_date(resource, current)
-        assert 'last_modified' in resource
-
-    def test_null_to_url(self):
-        resource = self.test_resource(self.URL_TYPE_LINK)
-        _update_resource_last_modified_date(resource=resource)
-        assert 'last_modified' in resource
-
-    def test_url_to_null(self):
-        resource = self.test_resource(self.URL_TYPE_LINK)
-        _update_resource_last_modified_date({}, resource)
-        assert 'last_modified' not in resource
-
-    def test_url_to_file(self):
-        current = self.test_resource(self.URL_TYPE_LINK)
-        resource = self.test_resource(self.URL_TYPE_FILE_UPLOAD)
-        _update_resource_last_modified_date(resource, current)
-        assert 'last_modified' in resource
+    @pytest.fixture
+    def resource_with_file(autouse=True):
```

Comment:
If you'd like to enable autouse it should go into decorator config:
```
@pytest.fixture(autouse=True)
def res_with_file():
[...]
```
I don't think it's a good idea to use autouse here anyway as you're not **always** using this fixtures in tests.
My suggestion: remove autouse kwarg.

Reply:
I didn't intend on using it originally, however without it I get the below error:
```
_____________ ERROR collecting ckanext/unaids/tests/test_plugin.py _____________
Could not determine arguments of <bound method ?.resource_with_file of <ckanext.unaids.tests.test_plugin.TestResourceLastModified object at 0x7fcd0c5da7d0>>: invalid method signature
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
```

Reply:
You're defining the method inside a class, so first argument needs to be `self`. You're just doing a really weird thing here which might be described like this:
```
class Foo():
   def bar(self=True):  <<<<<< def my_fixture(autouse=True)
         pass
```
There's no need to define the fixture in the scope of the class. Just move it up to the module level and you're good.